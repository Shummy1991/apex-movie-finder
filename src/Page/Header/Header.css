
h3 {
    text-align: center;
    color: white;
    font-size: 32px;
    font-variant: small-caps;
}

header {
    position: sticky;
    background-color: #0b1929;
    top: 0;
    padding-top: 4px;
    padding-bottom: 24px;
}
