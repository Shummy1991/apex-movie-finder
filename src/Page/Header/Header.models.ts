export interface IHeaderProps {
    handleSearch: (searchValue: string) => void;
    testID?: string;
    isLoading: boolean;
}
