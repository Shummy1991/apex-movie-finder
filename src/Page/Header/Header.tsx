import { Button, Grid, TextField } from "@mui/material";
import React, { useState } from "react";

import { IHeaderProps } from "./Header.models";
import "./Header.css";

const Header = ({ handleSearch, isLoading }: IHeaderProps) => {
    const [searchValue, setSearchValue] = useState("");
    const handleChangeSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
        setSearchValue(event.currentTarget.value);
    };
    const handleSubmit = () => {
        handleSearch(searchValue);
    };
    const handleKeyPress = (event: React.KeyboardEvent<HTMLDivElement>) => {
        if (searchValue.length === 0) {
            return;
        }
        if (event.key === 'Enter') {
            handleSubmit();
        }
    };
    return (
        <header>
            <h3>Movie finder</h3>
            <Grid container spacing={2}>
                <Grid item flex={1}>
                    <TextField
                        onChange={handleChangeSearch}
                        value={searchValue}
                        variant="filled"
                        label="Enter a movie title"
                        fullWidth
                        autoFocus
                        onKeyPress={handleKeyPress}
                        disabled={isLoading}
                    />
                </Grid>
                <Grid item alignItems="center" display="flex">
                    <Button
                        onClick={handleSubmit}
                        disabled={searchValue.length === 0 || isLoading}
                    >
                        Search
                    </Button>
                </Grid>
            </Grid>
        </header>
    );
};

export default Header;
