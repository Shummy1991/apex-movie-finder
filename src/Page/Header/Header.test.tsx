import "jest";
import { createEvent, fireEvent, render } from "@testing-library/react";

import Header from "./Header";

const defaultProps = {
    handleSearch: jest.fn(),
    isLoading: false,
};

describe("Header", () => {
    it("matches snapshot", () => {
        const { asFragment } = render(<Header {...defaultProps} />)
        expect(asFragment()).toMatchSnapshot();
    });
    it("input changes value on change", async () => {
        const { findAllByLabelText, asFragment } = render(<Header {...defaultProps} />)
        const [textField] = await findAllByLabelText("Enter a movie title");
        fireEvent.change(
            textField,
            createEvent("input", textField, {
                target: {
                    value: "asdf"
                }
            })
        );
        expect(asFragment()).toMatchSnapshot();
    });
    it("enter key press triggers submit", async () => {
        const { findAllByLabelText } = render(<Header {...defaultProps} />)
        const [textField] = await findAllByLabelText("Enter a movie title");
        fireEvent.keyPress(textField, { key: "Enter", code: "Enter", charCode: 13 });
        expect(defaultProps.handleSearch).not.toBeCalled();

        const value = "asdf";
        fireEvent.change(
            textField,
            createEvent("input", textField, {
                target: {
                    value,
                }
            })
        );
        fireEvent.keyPress(textField, { key: "A", code: "A", charCode: 97 });
        expect(defaultProps.handleSearch).not.toBeCalled();
        fireEvent.keyPress(textField, { key: "Enter", code: "Enter", charCode: 13 });
        expect(defaultProps.handleSearch).toBeCalledWith(value);
    });
});
