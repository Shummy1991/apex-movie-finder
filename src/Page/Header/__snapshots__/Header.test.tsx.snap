// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header input changes value on change 1`] = `
<DocumentFragment>
  <header>
    <h3>
      Movie finder
    </h3>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-1s76rlx-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-focused css-o943dk-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r1:"
            id=":r1:-label"
          >
            Enter a movie title
          </label>
          <div
            class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-gjbq6i-MuiInputBase-root-MuiFilledInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiFilledInput-input MuiInputBase-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
              id=":r1:"
              type="text"
              value="asdf"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-qh4ozh-MuiGrid-root"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Search
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
</DocumentFragment>
`;

exports[`Header matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <h3>
      Movie finder
    </h3>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-1s76rlx-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused css-o943dk-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r0:"
            id=":r0:-label"
          >
            Enter a movie title
          </label>
          <div
            class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-gjbq6i-MuiInputBase-root-MuiFilledInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiFilledInput-input MuiInputBase-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
              id=":r0:"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-qh4ozh-MuiGrid-root"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          disabled=""
          tabindex="-1"
          type="button"
        >
          Search
        </button>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
